src/main/java/com/uid2/admin/vertx/service/ServiceLinkService.java

```diff
@@ -107,6 +109,19 @@ private void handleServiceLinkAdd(RoutingContext rc) {
                 return;
             }
 
+            final Set<Role> roles;
+            try {
+                roles = rolesSpec.stream().map(s -> Role.valueOf((String) s)).collect(Collectors.toSet());
```

Comment:
We're validating the input for roles but not storing it? I thought this was going to be done in a separate ticket